### PR TITLE
feat: POFE-536 - allow PO user to specify his merchant ID for ApplePay

### PR DESCRIPTION
**Pull request by julien-rodrigues-cko**
## Description
The goal of this PR is to update the `ApplePay` session to allow user to specify an `ApplePay` Merchant ID. 

## Solution
- add a new field `merchantApplePayCertificateId` in the `ApplePayPaymentRequest`
- when making the `POST` to `/session` we want to specify in the API request the Merchant ID 
- when making the `POST` to `/card` we want to specify in the API request the Merchant ID  

## Jira Issue
[POFE-536](https://checkout.atlassian.net/browse/POFE-536)

### CODE REVIEW

File: src/processout/applepay.ts [hunk 1]
```diff
@@ -5,280 +5,312 @@
...
+      var t = this;
+      this.session.onvalidatemerchant = function (event: any): void {
+        t.instance.apiRequest(
+          "post",
+          t.instance.endpoint("api", "/applepay/sessions"),
+          {
+            session_url: event.validationURL,
+            domain_name: window.location.hostname,
+            merchant_certificate_id: t.request.merchantCertificateId,
+            applepay_mid: t.request.merchantApplePayCertificateId,
```
Comment (zhengyi-li-cko) [inline src/processout/applepay.ts:108]: Is this the only real change in this PR? It is quite hard to identify among the format change even filter out the whitespace.

Comment (zhengyi-li-cko) [reply src/processout/applepay.ts:108]: It would be nice to have 2 commits, separate the format changes from the actual code changes.

Comment (julien-rodrigues-cko) [reply src/processout/applepay.ts:108]: Yeah you right, here are the 3 changes: 

- https://github.com/processout/processout.js/pull/78/files#diff-9a80fcfc5f93fa79d9070aa264a6c84032056e11cbe3ae4bbead6ca8a80ca167R131
- https://github.com/processout/processout.js/pull/78/files#diff-9a80fcfc5f93fa79d9070aa264a6c84032056e11cbe3ae4bbead6ca8a80ca167R108
- https://github.com/processout/processout.js/pull/78/files#diff-3e255e8bd53882cb8469283995b924befca568fc11e5c8a0cabcccde6b360ee3R17

**Comment by zhengyi-li-cko** [comment]
Just trying to understand the bigger picture. I'm not sure the final result of having merchant id for ApplePay... For example, what will happen from merchant point of view after this change, is that merchant can't use ApplePay before and now they can use ApplePay?

**Comment by julien-rodrigues-cko** [comment]
> Just trying to understand the bigger picture. I'm not sure the final result of having merchant id for ApplePay... For example, what will happen from merchant point of view after this change, is that merchant can't use ApplePay before and now they can use ApplePay?

Basically, to process a payment using ApplePay you need to create your own Merchant ID from the Apple dev tool, and then you can go on the dashboard and follow the steps to branch your Merchant ID with your project. So initially, you were already able to do that and then to process a payment using ApplePay but you were only able to have one Merchant ID, so that answer your question, merchant was able to use ApplePay before this. Now, with an additional work on the back-end, the merchant can have multiple Merchant IDs set for a project, so with this work on `processout.js`, when the merchant will create the ApplePay session, he will be able to specify which Merchant ID he wants to use. 

I will try to plan a quick session with @adam-blevins-cko and you to show you what's happening, because it was not very clear for me at the beginning, so I assume it will be useful for you to know.

**Review by zhengyi-li-cko — APPROVED** [review]

**Pull request merged**